### PR TITLE
feat(@angular-ru/common): add mapObject function

### PR DESCRIPTION
**Pull request by Schirbak**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/common/integration/tests/object/object.spec.ts [hunk 1]
```diff
@@ -411,4 +412,27 @@ describe('[TEST]: Object', () => {
 
         expect(pathsOfObject(willem)).toEqual(['name', 'city']);
     });
+
+    it('should correct map objects by keys', function () {
```
Comment (splincode) [inline packages/common/integration/tests/object/object.spec.ts:416]: please, add test with deep objects

File: packages/common/object/src/map-object.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { PlainObjectOf } from '@angular-ru/common/typings';
+
+export function mapObject<Type, ReturnType>(
+    object: PlainObjectOf<Type>,
```
Comment (splincode) [inline packages/common/object/src/map-object.ts:4]: `PlainObjectOf<Type>` not `Type`?

File: packages/common/object/src/map-object.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    mapper: (obj: Type, key: string) => ReturnType
+): PlainObjectOf<ReturnType> {
+    const result: PlainObjectOf<ReturnType> = {};
+
+    for (const key in object) {
+        if (object.hasOwnProperty(key)) {
```
Comment (StepaZol) [inline packages/common/object/src/map-object.ts:10]: when can you get "false" here?

Comment (splincode) [reply packages/common/object/src/map-object.ts:10]: @StepaZol 

image

Comment (splincode) [reply packages/common/object/src/map-object.ts:10]: Looping over objects with a for in loop will include properties that are inherited through the prototype chain. This behavior can lead to unexpected items in your for loop.

image

Comment (splincode) [reply packages/common/object/src/map-object.ts:10]: https://eslint.org/docs/rules/guard-for-in

File: packages/common/object/src/map-object.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { PlainObjectOf } from '@angular-ru/common/typings';
+
+export function mapObject<Type, ReturnType>(
+    object: PlainObjectOf<Type>,
+    mapper: (obj: Type, key: string) => ReturnType
+): PlainObjectOf<ReturnType> {
+    const result: PlainObjectOf<ReturnType> = {};
```
Comment (splincode) [inline packages/common/object/src/map-object.ts:7]: @Schirbak I recommended use JSON.parse with JSON.stringify for mapping fields
analogue - https://github.com/Angular-RU/angular-ru-sdk/blob/master/packages/common/object/src/replace-with-null.ts

File: packages/common/object/src/map-object.ts (hunk 3 above)
Comment (StepaZol) [reply packages/common/object/src/map-object.ts:10]: oh yep. I forgot about it

Comment (Schirbak) [reply packages/common/object/src/map-object.ts:10]: when we got a key from prototype chain. 

ESLint: The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.(guard-for-in)

File: packages/common/object/src/map-object.ts (hunk 4 above)
Comment (Schirbak) [reply packages/common/object/src/map-object.ts:7]: this function is not for deep mapping

File: packages/common/object/src/map-object.ts (hunk 2 above)
Comment (Schirbak) [reply packages/common/object/src/map-object.ts:4]: shto

Comment (splincode) [reply packages/common/object/src/map-object.ts:4]: why `export function mapObject<T, K>(obj: PlainObject<T>)` not `export function mapObject<T, K>(obj: T)`

**Review by splincode — APPROVED** [review]

**Pull request merged**